### PR TITLE
Update zbus to v4

### LINKED ISSUE #71: New major zbus release 4.0
Just a heads up that you will need to update this crate to work with the new version. I know it was released just a few hours ago, so I don't want to seem impatient. This is just a heads up.

(I'm so looking forward to getting smaller binary sizes when I can avoid a ton of duplicated dependencies.)

### PR DESCRIPTION
**Pull request by liff**
Fixes #71. The required changes seem to be pretty minor, unless there’s something I missed.

### CODE REVIEW

**Comment by VorpalBlade** [comment]
I have tested this branch and it works correctly in my use case (https://github.com/VorpalBlade/chezmoi_modify_manager) but it seems like you need to fix some clippy warnings, as well as bump MSRV of this repo (probable locations this needs to be done: `Cargo.toml`, the CI workflow files, and somewhere in the docs probably?).

**Review by VorpalBlade — APPROVED** [review]
Apart from the MSRV and (unrelated to your changes, but needed to be fixed for CI to let it through) clippy warnings this looks good to me.

File: src/proxy/collection.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use zbus::{
-    dbus_proxy,
+    proxy,
```
Comment (complexspaces) [inline src/proxy/collection.rs:13]: Nit: Let's remove this import and qualify it instead for consistency with the other files.

File: src/util.rs [hunk 2]
```diff
@@ -148,7 +148,7 @@ fn handle_signal(signal: Completed) -> Result<zvariant::OwnedValue, Error> {
     if args.dismissed {
         Err(Error::Prompt)
     } else {
-        Ok(args.result.into())
+        Ok(zvariant::OwnedValue::try_from(args.result)?)
```
Comment (complexspaces) [inline src/util.rs:151]: Question: Can we do `.map_err(From::from)` or similar here instead to avoid the `Ok(?)` pattern?

**Review by complexspaces — CHANGES_REQUESTED** [review]
Thank you for working on this update. Do you mind rebasing this on `master` and updating the MSRV here to the lowest version that passes (I think `zbus` 4 is between 1.69 and 1.75)?

Once CI runs I think this looks good to go and can be the starting place for `secret-service` 4.0.

File: src/util.rs (hunk 2 above)
Comment (liff) [reply src/util.rs]: Seems to work, changed.

**Review by complexspaces — APPROVED** [review]
Thanks!

**Comment by complexspaces** [comment]
Released in [v4.0.0 on crates.io](https://crates.io/crates/secret-service/4.0.0). Thanks again!

**Pull request merged**